client/src/templates/Introduction/components/challenges.tsx

```diff
@@ -9,6 +9,8 @@ import GreenNotCompleted from '../../../assets/icons/green-not-completed';
 import GreenPass from '../../../assets/icons/green-pass';
 import { executeGA } from '../../../redux/actions';
 import { ChallengeWithCompletedNode } from '../../../redux/prop-types';
+import { SuperBlocks } from '../../../../../shared/config/superblocks';
+import { challengeTypes } from '../../../../../shared/config/challenge-types';
```

Comment:
```suggestion
import { challengeTypes } from '../../../../../shared/config/challenge-types';

const getStepNumber = (dashedName: string) => {
  // dashedName should be in the format 'step-1' or 'task-1'
  const match = dashedName.match(/-(\d+)/);
  return match ? match[1] : '';
};
```
It's a bit verbose, but I think it's clearer as a result.